project-base/storefront/components/Basic/GoogleMap/GoogleMap.tsx

```diff
@@ -0,0 +1,142 @@
+
+    const points = useMemo(
+        () =>
+            markers?.map((marker) => ({
+                type: 'Feature',
+                properties: {
+                    cluster: false,
+                    markerId: marker.identifier ?? marker.uuid,
+                },
+                geometry: {
+                    type: 'Point',
+                    coordinates: [
+                        marker.locationLongitude !== null ? parseFloat(marker.locationLongitude) : null,
+                        marker.locationLatitude !== null ? parseFloat(marker.locationLatitude) : null,
+                    ],
+                },
+            })) ?? [],
+        [markers],
+    );
+
+    const { clusters } = useSupercluster({
+        points,
+        zoom,
+        bounds,
+        options: { radius: 75, maxZoom: 20 },
+    });
+
+    const markerClickHandler = (id: string) => {
+        if (isDetail) {
+            return;
+        }
+        const newActiveMarkerId = activeMarkerId === id ? '' : id;
+        setActiveMarkerId(newActiveMarkerId);
+        activeMarkerHandler?.(newActiveMarkerId);
+    };
+
+    useEffect(() => {
+        if (!closeMarkers) {
+            return;
+        }
+        setActiveMarkerId('');
+    }, [closeMarkers]);
+
+    return (
+        <div className="h-full w-full" data-testid={TEST_IDENTIFIER}>
+            <GoogleMapReact
+                yesIWantToUseGoogleMapApiInternals
+                bootstrapURLKeys={{ key: publicRuntimeConfig.googleMapApiKey }}
+                defaultCenter={{ lat: defaultLat, lng: defaultLng }}
+                defaultZoom={defaultZoom}
+                options={{
+                    disableDoubleClickZoom: true,
+                    fullscreenControl: false,
+                    zoomControlOptions: { position: 1 },
+                }}
+                onChange={({ zoom, bounds }) => {
+                    setZoom(zoom);
+                    setBounds([bounds.nw.lng, bounds.se.lat, bounds.se.lng, bounds.nw.lat]);
+                }}
+            >
+                {clusters.map((cluster) => {
+                    const [longitude, latitude] = cluster.geometry.coordinates;
+                    const { cluster: isCluster, point_count: pointCount, markerId } = cluster.properties;
+                    const isActive = cluster.properties.markerId === activeMarkerId;
+
+                    if (isCluster) {
+                        return (
+                            <ClusterMarker key={cluster.id} lat={latitude} lng={longitude}>
+                                {pointCount}
+                            </ClusterMarker>
+                        );
+                    }
+
+                    return (
+                        <GoogleMapMarker
+                            key={markerId}
+                            isActive={isActive}
+                            isDetail={isDetail}
+                            lat={latitude}
+                            lng={longitude}
```

Comment:
You do not seem to use these props inside the `GoogleMapMarker` component.

Reply:
They are consumed by `GoogleMapReact` parent component, and I need to place them on some `div` element for the library to render them on the map. But I have refactored this and added a comment explaining why we need it.